lib/node_modules/@stdlib/repl/test/integration/test.completer_engine.js

```diff
@@ -0,0 +1,488 @@
+
+		// First completion should be highlighted:
+		t.strictEqual( actual[ 0 ], '\x1B[7mzzzab\x1B[27m', 'returns expected value' );
+		t.strictEqual( data[ data.length - 9 ], '\x1B[2A', 'returns expected value' ); // bring cursor back
+
+		// Current line is cleared and the first completion is inserted:
+		t.strictEqual( data[ data.length - 7 ], '\x1B[0K', 'returns expected value' );
+		t.strictEqual( data[ data.length - 6 ], 'var zzzab = 1;var zzzxy = 2;zzzab', 'returns expected value' );
+
+		// Check for completions after `right` arrow key:
+		actual = extractCompletions( data[ data.length - 5 ] );
+
+		// Second completion should be highlighted:
+		t.strictEqual( actual[ 1 ], '\x1B[7mzzzxy\x1B[27m', 'returns expected value' );
+		t.strictEqual( data[ data.length - 4 ], '\x1B[2A', 'returns expected value' ); // bring cursor back
+
+		// Current line is cleared and the second completion is inserted:
+		t.strictEqual( data[ data.length - 2 ], '\x1B[0K', 'returns expected value' );
+		t.strictEqual( data[ data.length - 1 ], 'var zzzab = 1;var zzzxy = 2;zzzxy', 'returns expected value' );
+
+		t.end();
+	}
+});
+
+tape( 'a REPL instance supports bringing back the original line upon navigating back up from the TAB completions', function test( t ) {
+	var istream;
+	var opts;
+	var r;
+
+	istream = new DebugStream({
+		'name': 'repl-input-stream'
+	});
+	opts = {
+		'input': istream,
+		'settings': defaultSettings(),
+		'tty': {
+			'rows': 100,
+			'columns': 80
+		}
+	};
+	r = repl( opts, onClose );
+
+	// Declare variables with unique names in order to prevent namespace collisions:
+	istream.write( 'var zzabc = 1;' );
+	istream.write( 'var zzpqr = 2;' );
+
+	// Write the common beginning of the variable names in order to generate TAB completions:
+	istream.write( 'zz' );
+
+	// Write TAB to display completions:
+	istream.write( '\t' );
+
+	// Navigate down using down arrow:
+	istream.write('\u001B[B');
+
+	// Navigate up towards the line to bring back the original line:
+	istream.write('\u001B[A');
+
+	// Close the input stream:
+	istream.end();
+
+	// Close the REPL:
+	r.close();
+
+	function onClose( error, data ) {
+		if ( error ) {
+			t.fail( error.message );
+			return;
+		}
+
+		// Current line is cleared and the original line is inserted:
+		t.strictEqual( data[ data.length-2 ], '\x1B[0K', 'returns expected value' );
+		t.strictEqual( data[ data.length-1 ], 'var zzabc = 1;var zzpqr = 2;zz', 'returns expected value' );
+
+		t.end();
+	}
+});
+
+tape( 'a REPL instance supports displaying highlighted argument TAB completions', function test( t ) {
+	var istream;
+	var opts;
+	var r;
+
+	istream = new DebugStream({
+		'name': 'repl-input-stream'
+	});
+	opts = {
+		'input': istream,
+		'settings': defaultSettings(),
+		'tty': {
+			'rows': 100,
+			'columns': 80
+		}
+	};
+	r = repl( opts, onClose );
+
+	// Write the common beginning of the settings in order to generate TAB completions:
+	istream.write( 'settings(\'auto' );
```

Comment:
This test is likely to be brittle. If, in the future, we decide to rename the `autoPage` setting, this will result in a mysterious test failure unrelated to that change. Is there another way we can test this?